include/seqan3/io/sequence_file/format_sam.hpp

```
@@ -0,0 +1,346 @@
+              detail::consume(stream_view | ranges::view::take_while(!is_tab));
+            }
+            else
+            {
+              ranges::copy(stream_view | view::take_until_or_throw(is_tab)
+                                       | view::char_to<value_type_t<id_type>>,
+                                   ranges::back_insert_iterator{id});
+            }
+
+          }
+          else
+          {
+            detail::consume(stream_view | view::take_line_or_throw);
+          }
+
+          // Jump over all parts we are not interested in
+          detail::consume(stream_view | ranges::view::take_while(is_tab));
+          detail::consume(stream_view | ranges::view::take_while(!is_tab));
+          detail::consume(stream_view | ranges::view::take_while(is_tab));
+          detail::consume(stream_view | ranges::view::take_while(!is_tab));
+          detail::consume(stream_view | ranges::view::take_while(is_tab));
+          detail::consume(stream_view | ranges::view::take_while(!is_tab));
+          detail::consume(stream_view | ranges::view::take_while(is_tab));
+          detail::consume(stream_view | ranges::view::take_while(!is_tab));
+          detail::consume(stream_view | ranges::view::take_while(is_tab));
+          detail::consume(stream_view | ranges::view::take_while(!is_tab));
+          detail::consume(stream_view | ranges::view::take_while(is_tab));
+          detail::consume(stream_view | ranges::view::take_while(!is_tab));
+          detail::consume(stream_view | ranges::view::take_while(is_tab));
+          detail::consume(stream_view | ranges::view::take_while(!is_tab));
+          detail::consume(stream_view | ranges::view::take_while(is_tab));
+          detail::consume(stream_view | ranges::view::take_while(!is_tab));
+          detail::consume(stream_view | ranges::view::take_while(is_tab));
+
+
+
+          auto seq_view = stream_view | ranges::view::remove_if(is_free || is_digit)   // ignore whitespace and numbers
```

@h-2 I see that skipping whitespaces and numbers is also done when reading fasta/fastq so I am adressing you: Shouldn't the format rather throw an invalid-char error when a number occurrs whithin the sequence? 
@MitraDarja you commented to skip whitespaces but `is_free` checks (only) for `\r`? (I am also not sure if whitespaces should be allowed in SAM format)